Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt

```diff
@@ -310,20 +321,42 @@ private fun processCLib(args: Array<String>, additionalArgs: Map<String, Any> =
             argsToCompiler(staticLibraries, libraryPaths)
         }
         is StubIrDriver.Result.Metadata -> {
-            val moduleName = File(cinteropArguments.output).nameWithoutExtension
             val args = LibraryCreationArguments(
+                    metadata = stubIrOutput.metadata,
                     nativeBitcodePath = nativeOutputPath,
                     target = tool.target,
                     moduleName = moduleName,
                     outputPath = cinteropArguments.output,
-                    manifest = def.manifestAddendProperties
+                    manifest = def.manifestAddendProperties,
+                    dependencies = allLibraryDependencies
             )
             createInteropLibrary(args)
             return null
         }
     }
 }
 
+private fun resolveDependencies(
+        cinteropArguments: CInteropArguments, target: KonanTarget, mode: GenerationMode
+): List<KotlinLibrary> {
+    val libraries = cinteropArguments.library
+    val repos = cinteropArguments.repo
+    val noDefaultLibs = cinteropArguments.nodefaultlibs || cinteropArguments.nodefaultlibsDeprecated
+    val noEndorsedLibs = cinteropArguments.noendorsedlibs
+    val resolver = defaultResolver(
+            repos,
+            libraries.filter { it.contains(org.jetbrains.kotlin.konan.file.File.separator) },
+            target,
+            Distribution()
+    ).libraryResolver()
+    return resolver.resolveWithDependencies(
+            libraries.toUnresolvedLibraries,
+            noStdLib = mode == GenerationMode.SOURCE_CODE,
```

Comment:
Why so?

Reply:
I tried to preserve the old behaviour for `-mode sourcecode` when moved resolving process inside this file. One change at a time :)
Indeed, it can be `true` for both modes.